test/offset_calculator_test.rb

```diff
@@ -10,68 +11,68 @@ def test_it_exists
     assert_instance_of OffsetCalculator, oc
   end
 
-  def test_date_squared
-    oc = OffsetCalculator.new
-    date = "101517"
-
-    result = oc.date_squared(date)
+  def test_any_date_squared
+    kg = KeyGenerator.new
+    oc = OffsetCalculator.new(kg, Date.new(2015, 11, 12))
+    result = oc.date_squared
 
-    assert_equal 10305701289, result
+    assert_equal 14668843225, result
   end
 
-  def test_offset
-    oc = OffsetCalculator.new
-    date = "101517"
+  def test_todays_date_squared
+    kg = KeyGenerator.new
+    oc = OffsetCalculator.new(kg, Date.today)
+    result = oc.date_squared
 
-    result = oc.date_squared(date)
-
-    assert_equal 4, oc.offset.length
-    assert_equal "1289", oc.offset.join
+    assert_equal 25926474289, result
   end
 
-# ask instructor if this needs to happen
-  # def test_another_date_works
-  #   oc = OffsetCalculator.new
-  #   date = "053117"
-  #
-  #   result = oc.date_squared(date)
-  #
-  #   assert_equal 2821415689, result
-  # end
+  def test_date_array
+    kg = KeyGenerator.new
+    oc = OffsetCalculator.new(kg, Date.today)
+    result = oc.date_array
 
-  def test_if_a_offset_grabs_index_zero
-    oc = OffsetCalculator.new
-    date = "101517"
+    assert_equal 4, oc.date_array.length
+    assert_equal "4289", oc.date_array.join
+  end
 
-    result = oc.offset
+  def test_if_a_offset_grabs_index_zero
```

Comment:
these tests seem duplicative of the above test that checks for "4289", you probably could lose them.